src/shop/anya.py

```diff
@@ -148,10 +148,21 @@ def shop_loop(self):
                 claw_pos = []
                 img = grab().copy()
                 claw_keys = ["CLAW1", "CLAW2", "CLAW3"]
-                for ck in claw_keys:
-                    template_match = TemplateFinder(True).search(ck, img, roi=self.roi_vendor)
+
+                # new approach iterates on image until all matches are added to the list
+                claws_remaining = True
+                claws_img = img.copy() # make an additional copy to avoid breaking usage of 'img' at later time
+                while claws_remaining:
+                    template_match = TemplateFinder(True).search(claw_keys, claws_img, roi=self.roi_vendor, best_match=True)
                     if template_match.valid:
                         claw_pos.append(template_match.center)
+                        # black out rectangle from image for future iterations
```

Comment:
I like your approach here, and this is what we're working to implement in other PR's. There is a function in utils/misc.py that could potentially simplify some of the logic here-- mask_by_roi(). I think claws_img = mask_by_roi(claws_image, template_match.region, "inverse") might do it. I'd like to give #683 priority since if it works we'll just have to refactor this code anyway, but if it takes a while I'd be ok with this as a temporary solution. My apologies it took a while to review this.

Reply:
Thanks for the review -- I totally overlooked `mask_by_roi()`! I can implement and test that in the next couple days in case you do decide to use this as a temp solution.